vortex-array/src/arrays/primitive/compute/is_constant.rs

```diff
@@ -9,24 +11,49 @@ impl IsConstantFn<&PrimitiveArray> for PrimitiveEncoding {
         &self,
         array: &PrimitiveArray,
         _opts: &IsConstantOpts,
-    ) -> vortex_error::VortexResult<Option<bool>> {
-        let is_constant = match_each_native_ptype!(array.ptype(), |$P| {
-            compute_is_constant(array.as_slice::<$P>())
+    ) -> VortexResult<Option<bool>> {
+        let is_constant = match_each_native_ptype!(array.ptype(), integral: |$P| {
+            compute_is_constant::<_, {256 / size_of::<$P>()}>(array.as_slice::<$P>())
```

Comment:
Is this attempting to convince the compiler to SIMD? It's a weird intermediate between total auto-vectorizing and explicit std::simd

Reply:
I think chunking/tiling it a very well known optimisation

Reply:
This is an attempt to autovectorize. Do you think std::simd would be better?

Reply:
my 2c - this is much nicer, no nightly features (`array_chunks` seems very close to stabilization) and much easier to maintain.